packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -131,57 +131,7 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 
 	return useMemo(
 		() => ( {
-			...Object.fromEntries(
-				Object.entries( settings ).filter( ( [ key ] ) =>
```

Comment:
I see this has been removed, where's the filtering happening now? (Sorry I missed the ping earlier)

Reply:
There's no filtering now, the entire editor settings object will be propagated to block editor settings. Technically it causes no harm, or does it? I can see one good reason for filtering, and that is a reliable and well-defined public API, but that's all.

I removed the filtering because it was standing in the way of synchronizing settings in the mobile `EditorProvider`. Before this PR, it called `dispatch( 'core/block-editor').updateSettings` directly, and circumvented the filter. But now it calls `dispatch( 'core/editor' ).updateEditorSettings` and relies on the settings to be synced to block editor. Here the updates go through the filter.

I found it hard to figure out the set of supported mobile block editor settings, see e.g. the `capabilities` confusion solved in #47417. Once all this is sorted out, I'll be happy to add the filter back.

Reply:
There's no filtering now, the entire editor settings object will be propagated to block editor settings. Technically it causes no harm, or does it?

Well now, folks will be able to access settings using `select( 'core/block-editor' ).getSettings() that were specific to the edit-post package and it creates wrong expectations. Ideally only block-editor specific settings should be passed down.

Reply:
I think the important thing here is that the filtering is going to be needed for 6.2 (so before next Gutenberg RC) if we don't backward compatibility issues.

Reply:
OK I'll add it back ASAP. Note that it's possible to circumvent this filter and set an arbitrary block-editor setting with `dispatch( 'core/block-editor').updateSettings()`. Mobile editor does that with `capabilities` and other settings: `capabilities` is read by the `editor` package, by specific blocks ("image" and "missing") and by `rich-text` (to implement support for mentions and xposts, which seems Jetpack or A8c-specific to me?)

Reply:
yeah, I mean folks might ask us why `enableCustomFields` for instance is not available anymore in the "site editor" or something while it's an edit-post specific flag...

Granted that it's not perfect, I guess I'm just leaning towards a defensive mode here in terms of what's exposed and where.